source/common/common/logger.h

```diff
@@ -123,48 +124,51 @@ template <Id id> class Loggable {
 
 } // Logger
 
+/**
+ * Base logging macros.
+ */
 #ifdef NDEBUG
-#define log_trace(...)
-#define log_debug(...)
+#define log_trace(LOGGER, ...)
+#define log_debug(LOGGER, ...)
 #else
-#define log_trace(...) log().trace(__VA_ARGS__)
-#define log_debug(...) log().debug(__VA_ARGS__)
+#define log_trace(LOGGER, ...) LOGGER.trace(__VA_ARGS__)
+#define log_debug(LOGGER, ...) LOGGER.debug(__VA_ARGS__)
 #endif
 
+#define log_info(LOGGER, ...) LOGGER.info(__VA_ARGS__)
+#define log_warn(LOGGER, ...) LOGGER.warn(__VA_ARGS__)
+#define log_err(LOGGER, ...) LOGGER.err(__VA_ARGS__)
+#define log_critical(LOGGER, ...) LOGGER.critical(__VA_ARGS__)
+
 /**
- * Convenience macros for logging with connection ID.
+ * Convenience macros to call the above macros with or without a logger.
  */
-#define conn_log(LOG, LEVEL, FORMAT, CONNECTION, ...)                                              \
-  LOG.LEVEL("[C{}] " FORMAT, (CONNECTION).id(), ##__VA_ARGS__)
+#define log_facility(LEVEL, ...) log_##LEVEL(log(), ##__VA_ARGS__)
```

Comment:
Totally agree.  This name was due to many name collisions (outside of just the Loggable class) with the macro `log`.  I was also tossing around `class_log` (also ambiguous and verbose) or `lg` (which was inconsistent with the other names, but was nice in that it kept the preferred macro short).  For now, I can add some documentation for what a facility is, and I'll ponder on the name.